_posts/2018-08-30-fullstack-typed-api-with-typescript.md

```diff
@@ -0,0 +1,428 @@
+---
+layout: post
+title: Fullstack shared typed api with TypeScript
+author: jgke
+excerpt: >
+  TypeScript makes it possible to share type information between frontend and
+  backend. Here's how to use it to ensure consistent api calls between the
+  frontend and backend.
+tags:
+ - TypeScript
+ - Full stack
+---
+
+Got a yet another error due to the JavaScript frontend slipping an unnoticed
+``undefined`` to a request or passing a string instead of a number, which
+causes requests to fail without a clear reason (or worse, succeed)? Maybe you
```

Comment:
or even worse. Ottaisin myös sulkeet pois. ...eithout a clear reason. Or even worse, succeed in weird way. Tai jotain sinnepäin.